lib/report/k8s.go

```diff
@@ -75,17 +100,28 @@ func capturePreviousContainerLogs(ctx context.Context, namespaces []string, runn
 				continue
 			}
 			for _, container := range containers {
-				// Collect logs for the previous instance of the container if there's any.
-				name := fmt.Sprintf("k8s-logs-%v-%v-%v-prev", namespace, pod, container)
-				collectors = append(collectors, Cmd(name, kubectl.Command("logs", pod,
-					"--namespace", namespace, "-p",
-					fmt.Sprintf("-c=%v", container)).Args()...))
+				collectors = append(collectors, containerLogCollector(namespace, pod, container, since, previous))
 			}
 		}
 	}
 	return collectors
 }
 
+// containerLogCollector returns a k8s log collector with the provided values
+// and configuration. If previous is true, only collect logs for previously
+// running instances of the container.
+func containerLogCollector(namespace, pod, container string, since time.Duration, previous bool) (cmd Command) {
+	name := fmt.Sprintf("k8s-logs-%v-%v-%v", namespace, pod, container)
+	if previous {
+		name += "-prev"
+	}
+	return Cmd(name, kubectl.Command("logs", pod,
+		"--namespace", namespace,
+		"--since", since.String(),
```

Comment:
What if "since" is 0? Will it work as expected, i.e. collect all logs?

Reply:
Yup, default is `0s`

`--since=0s: Only return logs newer than a relative duration like 5s, 2m, or 3h. Defaults to all logs. Only one of
since-time / since may be used.`